openff/toolkit/typing/engines/smirnoff/forcefield.py

```diff
@@ -1505,6 +1505,40 @@ def _get_parameter_handler_class(self, tagname):
             raise KeyError(msg)
         return ph_class
 
+    def get_partial_charges(self, molecule, **kwargs):
+        """Generate the partial charges for the given molecule in this force field.
+
+        As any `ParameterHandler` may in principle modify charges, the entire
+        force field must be applied to the molecule to produce the charges.
+        Calls to this method from incorrectly or incompletely specified `ForceField`
+        objects thus may raise an exception.
+
+        Parameters
+        ----------
+        molecule : openff.toolkit.topology.Molecule
+            The ``Molecule`` corresponding to the system to be parameterized
+        toolkit_registry : openff.toolkit.utils.toolkits.ToolkitRegistry, optional. default=GLOBAL_TOOLKIT_REGISTRY
+            The toolkit registry to use for operations like conformer generation and
+            partial charge assignment.
+
+        Returns
+        -------
+        charges : simtk.unit.Quantity with shape (n_atoms,) with units compatible with elemental charge
+            The partial charges of the provided molecule in this force field.
+
+        """
+        _, top_with_charges = self.create_openmm_system(
+            molecule.to_topology(), return_topology=True, **kwargs
+        )
+
+        if top_with_charges.n_topology_virtual_sites != 0:
+            raise ValueError(
```

Comment:
I'd like us to move away from re-using built-in exceptions, so let's replace this with a custom exception. You can find a template for doing this at the top of the file.

Reply:
Done in 43a27a6